### PR TITLE
Refactor state handling

### PR DESCRIPTION
**Pull request by calmh**
Move state tracking into the puller/scanner objects. This is a first step towards resolving #1391.
    
Rename Puller and Scanner to roFolder and rwFolder as they have more duties than just pulling and scanning, and don't need to be exported.

The second commit is just the file rename, so that the first commit contains a usable diff.

### CODE REVIEW

**Comment by calmh** [comment]
@st-jenkins run integration tests please

File: internal/model/model.go [hunk 1]
```diff
@@ -180,18 +152,7 @@ func (m *Model) StartFolderRW(folder string) {
...
-		scanIntv:        time.Duration(cfg.RescanIntervalS) * time.Second,
-		model:           m,
-		ignorePerms:     cfg.IgnorePerms,
-		lenientMtimes:   cfg.LenientMtimes,
-		progressEmitter: m.progressEmitter,
-		copiers:         cfg.Copiers,
-		pullers:         cfg.Pullers,
-		queue:           newJobQueue(),
-	}
+	p := newRWFolder(m, folder, time.Duration(cfg.RescanIntervalS)*time.Second, cfg)
```
Comment (AudriusButkevicius) [inline internal/model/model.go]: Given we are passing in the config (we should pass the wrapper ideally), can't we just  check cfg.RescanIntervalS inside?

File: internal/model/folderstate.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+	mut     sync.Mutex
+	current folderState
+	changed time.Time
+}
+
+func (s *stateTracker) setState(newState folderState) {
+	s.mut.Lock()
+	if newState != s.current {
+		/* This should hold later...
```
Comment (AudriusButkevicius) [inline internal/model/folderstate.go]: I guess you should just uncomment this to catch the bugs.

**Comment by AudriusButkevicius** [comment]
It does look like a refactor hence feels safe.
The test failure is a bit random though.

File: internal/model/model.go (hunk 1 above)
Comment (calmh) [inline internal/model/model.go]: Agreed, will fix.

File: internal/model/folderstate.go (hunk 2 above)
Comment (calmh) [inline internal/model/folderstate.go]: Yep, I intend to, in the next step. :)

**Comment by calmh** [comment]
Ah. The test failure

```
--- FAIL: TestIgnores (1.89s)
	ignore_test.go:90: Incorrect number of files after initial scan, 0 != 24
```

is actually genuine. With this change we can't accept scan requests from the outside before the folder is started by `StartFolderR[OW]`, so probably the test doesn't do that. Will fix.

**Comment by calmh** [comment]
@st-jenkins run integration tests please

**Comment by calmh** [comment]
@st-jenkins run integration tests please

**Comment by AudriusButkevicius** [comment]
So it's sad that git doesn't do revisions, so I could see where the issues with the tests was.

**Comment by calmh** [comment]
Indeed. That's why some prefer gerrit I hear... But https://github.com/calmh/syncthing/commit/62ef128f0741d7478a72ec86f44fc77463a97418?diff=unified#diff-1720cf6fe63c017f0aa2e576e128f6b2L74 was that thing.

**Pull request merged**